Vizzuality/heco-invest
# fix: Approve/Reject action at backoffice uses account language

Fixing of backoffice bug which blocked successful approving/rejecting of record which had empty string at attribute that is mandatory, but fallback string for record default language existed at same time. Approving/rejecting of Investor/PD should be done with respect to `language` so validation error is not raised in such cases.

**EDIT:** Further investigation uncovered a bigger problem where translated attributes always fallback to `I18n.default_locale` which is wrong. I have added fix which makes sure that fallback is always done to record `language`. This ensures that validations are always run for correct language and also correct strings are returned via API.

I am also removing `language` from Investor and PD tables, because Account already contains `language` and I am taking its value from there --> I have discovered several weird situations where Account and Investor/PD language were different which should never happen.

## Testing instructions

Create Investor or PD which have required attribute empty at language which is not selected language of record. For example:
```
I18n.with_locale "pt" do
   unapproved_investor.account.update! language: "pt", about_pt: "ABOUT PT", about_en: ""
end
```
You should be able to approve/reject such record when Language of Backoffice is set up to `EN` <-- `about_en` is empty, but fallback data (`about_pt`) exists, therefore no validation error should be raised.

## Tracking

BugFix: https://vizzuality.atlassian.net/browse/LET-806
